core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -101,100 +103,149 @@ type EthBroadcaster struct {
-		triggers:         triggers,
-		chStop:           make(chan struct{}),
-		wg:               sync.WaitGroup{},
+		initSync:         sync.Mutex{},
+		isStarted:        false,
 		autoSyncNonce:    autoSyncNonce,
 	}
 }
 
 // Start starts EthBroadcaster service.
 // The provided context can be used to terminate Start sequence.
-func (eb *EthBroadcaster) Start(ctx context.Context) error {
+func (eb *EthBroadcaster[ADDR, TX_HASH, BLOCK_HASH]) Start(_ context.Context) error {
 	return eb.StartOnce("EthBroadcaster", func() (err error) {
-		eb.ethTxInsertListener, err = eb.eventBroadcaster.Subscribe(pg.ChannelInsertOnEthTx, "")
-		if err != nil {
-			return errors.Wrap(err, "EthBroadcaster could not start")
-		}
+		return eb.startInternal()
+	})
+}
 
-		eb.wg.Add(len(eb.addresses))
-		for _, k := range eb.addresses {
-			triggerCh := make(chan struct{}, 1)
-			eb.triggers[k] = triggerCh
-			go eb.monitorEthTxs(k, triggerCh)
-		}
+// startInternal can be called multiple times, in conjunction with closeInternal. The TxMgr uses this functionality to reset broadcaster multiple times in its own lifetime.
+func (eb *EthBroadcaster[ADDR, TX_HASH, BLOCK_HASH]) startInternal() error {
+	eb.initSync.Lock()
+	defer eb.initSync.Unlock()
+	if eb.isStarted {
+		return errors.New("EthBroadcaster is already started")
+	}
+	var err error
+	eb.ethTxInsertListener, err = eb.eventBroadcaster.Subscribe(pg.ChannelInsertOnEthTx, "")
+	if err != nil {
+		return errors.Wrap(err, "EthBroadcaster could not start")
+	}
+	eb.enabledAddresses, err = eb.ks.EnabledAddressesForChain(&eb.chainID)
+	if err != nil {
+		return errors.Wrap(err, "Broadcaster: failed to load EnabledAddressesForChain")
+	}
 
-		eb.wg.Add(1)
-		go eb.ethTxInsertTriggerer()
+	if len(eb.enabledAddresses) > 0 {
+		eb.logger.Debugw(fmt.Sprintf("Booting with %d keys", len(eb.enabledAddresses)), "keys", eb.enabledAddresses)
+	} else {
+		eb.logger.Warnf("Chain %s does not have any eth keys, no transactions will be sent on this chain", eb.chainID.String())
+	}
+	eb.chStop = make(chan struct{})
+	eb.wg = sync.WaitGroup{}
+	eb.wg.Add(len(eb.enabledAddresses))
+	eb.triggers = make(map[string]chan struct{})
+	for _, addr := range eb.enabledAddresses {
+		triggerCh := make(chan struct{}, 1)
+		eb.triggers[addr.String()] = triggerCh
+		go eb.monitorEthTxs(addr, triggerCh)
+	}
 
-		return nil
-	})
+	eb.wg.Add(1)
+	go eb.ethTxInsertTriggerer()
+
+	eb.isStarted = true
+	return nil
 }
 
 // Close closes the EthBroadcaster
-func (eb *EthBroadcaster) Close() error {
+func (eb *EthBroadcaster[ADDR, TX_HASH, BLOCK_HASH]) Close() error {
 	return eb.StopOnce("EthBroadcaster", func() error {
-		if eb.ethTxInsertListener != nil {
-			eb.ethTxInsertListener.Close()
-		}
-
-		close(eb.chStop)
-		eb.wg.Wait()
-		return nil
+		return eb.closeInternal()
 	})
 }
 
-func (eb *EthBroadcaster) Name() string {
+func (eb *EthBroadcaster[ADDR, TX_HASH, BLOCK_HASH]) SetIsUnitTest() {
+	eb.isUnitTestInstance = true
+}
```

Comment:
having this function feels hacky - the tests worked fine with call process unstarted beforehand. 

i feel like the service itself shouldn't be handling test scenarios, that should be either happening in the test/mock dependencies that are injected